model_analyzer/config/input/config_command_profile.py

```diff
@@ -1529,19 +1529,34 @@ def _autofill_values(self):
 
             # Run parameters
             if not model.parameters():
-                new_model["parameters"] = {
-                    "batch_sizes": self.batch_sizes,
-                    "concurrency": self.concurrency,
-                    "request_rate": self.request_rate,
-                }
+                if self.run_config_search_mode != "optuna":
+                    new_model["parameters"] = {
+                        "batch_sizes": self.batch_sizes,
+                        "concurrency": self.concurrency,
+                        "request_rate": self.request_rate,
+                    }
+                else:
+                    if self._fields["batch_sizes"].is_set_by_user():
```

Comment:
Why does this work differently than concurrency and request_rate? Can we remove the branching to speed up the code? E.g. set self.batch_sizes to an empty list by default then update them if the user sets batch sizes?

Reply:
Actually, it looks like Python may not optimize for branchless code, so removing the branching should only be done if it makes the code cleaner without making it do more work (I think it still might).

StackOverflow conversation on why Python does not optimize branchless code [here](stackoverflow.com/questions/72118249). Note that I'm seeing different things online about whether it optimizes for branchless programming or not.

Reply:
Understood, thanks for sharing the greater code logic. I'll resolve this.

If it's something you're interested, a ternary operator could look quite nice in a situation like this. That's a personal choice though, this looks good too. `new_model["parameters"] = self._fields["batch_sizes"].is_set_by_user() ? self.batch_sizes : []`